### PR TITLE
Adds implementation and test for API: `/v1/catalogs`

### PR DESCRIPTION
**Pull request by PuneetPunamiya**
- This API will find and return an array of all catalogs
    with id, name and type as information of catalog

Signed-off-by: Puneet Punamiya <ppunamiy@redhat.com>

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

### CODE REVIEW

File: api/pkg/service/catalog/catalog.go [hunk 1]
```diff
@@ -78,3 +80,25 @@ func (s *service) Refresh(ctx context.Context, p *catalog.RefreshPayload) (*cata
...
+		log.Error(err)
+		return nil, internalError
+	}
+
+	res := &catalog.ListResult{
+		Data: []*catalog.Catalog{},
+	}
+
+	for _, c := range all {
+		res.Data = append(res.Data, &catalog.Catalog{ID: c.ID, Name: c.Name, Type: c.Type})
```
Comment (SM43) [inline api/pkg/service/catalog/catalog.go:100]: may be we should add url also?

Comment (SM43) [reply api/pkg/service/catalog/catalog.go:100]: if someone is integrating APIs then knowing which catalogs are in hub this API would be useful
I think we should expose it for users.. wdyt

Comment (piyush-garg) [reply api/pkg/service/catalog/catalog.go:104]: Yes, makes sense

**Comment by SM43** [comment]
If we are going to support `/v1/catalogs` then we don't need `/catalogs`  hub UI can use the v1. so i think we should remove `/catalogs`.
wdyt?

File: api/v1/design/catalog.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+var _ = Service("catalog", func() {
+	Description("The Catalog Service exposes endpoints to interact with catalogs")
+
+	HTTP(func() {
+		Path("/v1")
+	})
+
+	Error("internal-error", ErrorResult, "Internal Server Error")
+	Error("not-found", ErrorResult, "Resource Not Found Error")
```
Comment (SM43) [inline api/v1/design/catalog.go:30]: is not-found required?

File: api/v1/design/catalog.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+package design
+
+import (
+	types "github.com/tektoncd/hub/api/design/types"
+	. "goa.design/goa/v3/dsl"
+)
+
+var _ = Service("catalog", func() {
+	Description("The Catalog Service exposes endpoints to interact with catalogs")
```
Comment (SM43) [inline api/v1/design/catalog.go:23]: interact with catalogs? may be something else

Comment (PuneetPunamiya) [reply api/v1/design/catalog.go:23]: Apologies will change the description

Comment (SM43) [reply api/v1/design/catalog.go:23]: apologies 😄

File: api/v1/design/catalog.go [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+package design
+
+import (
+	types "github.com/tektoncd/hub/api/design/types"
+	. "goa.design/goa/v3/dsl"
+)
+
+var _ = Service("catalog", func() {
+	Description("List of catalogs")
```
Comment (sm43) [inline api/v1/design/catalog.go:23]: `The catalog service provides details about catalogs.`

**Comment by sm43** [comment]
It seems there is a bug in goa, using `view` in the catalog generates code that fails the test in CLI.
the generated code in `gen/resource/service.go ` 
```
// NewViewedResources initializes viewed result type Resources from result type
// Resources using the given view.
func NewViewedResources(res *Resources, view string) *resourceviews.Resources {
	p := newResourcesView(res)
	return &resourceviews.Resources{Projected: p, View: "default"}
}
```
 this is used in cli to generated responses for API. but it si not considering the view and using the `default` view always
 could you please file a bug?

File: api/v1/service/catalog/catalog.go [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+		log.Error(err)
+		return nil, internalError
+	}
+
+	res := &catalog.ListResult{
+		Data: []*catalog.Catalog{},
+	}
+
+	for _, c := range all {
+		res.Data = append(res.Data, &catalog.Catalog{ID: c.ID, Name: c.Name, Type: c.Type, URL: &c.URL})
```
Comment (pratap0007) [inline api/v1/service/catalog/catalog.go:55]: could you please add `URL`  in required section of `type.go`

File: api/design/types/type.go [hunk 6]
```diff
@@ -50,7 +50,7 @@ var Category = Type("Category", func() {
 	Required("id", "name", "tags")
 })
 
-var Catalog = Type("Catalog", func() {
+var Catalog = ResultType("Catalog", func() {
```
Comment (pratap0007) [inline api/design/types/type.go:53]:
```suggestion
var Catalog = ResultType("application/vnd.hub.catalog","Catalog", func() {
```

Comment (pratap0007) [inline api/design/types/type.go:53]:
```suggestion
var Catalog = ResultType("application/vnd.hub.catalog","Catalog", func() {
```

**Pull request merged**